reference/interbit-ui-components/CONTRIBUTING.adoc

```diff
@@ -0,0 +1,205 @@
+= Contributing
+
+This document is for developers who wish to contribute to the
+`interbit-ui-components` package within the `interbit` repo. For general
+information on contributing to
+link:https://github.com/interbit/interbit[Interbit], please see this
+link:https://github.com/interbit/interbit/blob/master/CONTRIBUTING.md[Contribution Guide].
+
+
+== Creating components
+
+=== File structure
+
+[cols="1a,1a", options="header"]
+|===
+| Directory or file
+| Description
+
+
+| `src/assets`
+| Location for image assets.
+
+| `src/components`
+| React components exported from this package.
+
+| `src/components/BlockExplorer`
+| An interactive component displaying a chain's state, its blocks, and actions
+dispatched.
+
+| `src/components/Covenant`
+| A set of forms to dispatch chain actions. It is wrapped by `LinkedCovenant`
+and uses `FormTable` components.
+
+| `src/components/Navigation`
+| A set of navigation components such as header, footer, and sidebar menus.
+
+| `src/components/ReduxForm`
+| Helper functions for creating and validating Redux forms.
+
+| `src/components/UIKit`
+| A set of simple UI components.
+
+| `src/help`
+| Helper functionality for Redux forms, used with the `EditableTable`
+component. This directory is flagged for refactoring.
+
+| `src/scss/bootstrap`
+| SCSS files from React Bootstrap. These files should not be edited.
+
+| `src/scss/interbit`
+| SCSS files for Interbit styles. Each component has its own `scss` file which
+is imported in `interbit.scss`. Style overrides for React Bootstrap components
+are also located in this folder.
+
+| `src/tests/exports.tests.js`
+| A set of simple tests to check that exported components exist.
+
+| `src/tests/uiKit.test.js`
+| Rendering tests for UI Kit components.
+
+| `src/styleguide`
+| Build files for React Styleguidist.
+
+| `index.js`
+| Export statements for components.
+
+| `styleguide.config.js`
+| Configuration file for React Styleguidist.
+
+
+=== Code structure
+
+We are in the process of defining the code structure for
+Interbit UI Components. At this time, we ask that contributing developers use
+the existing code as an example of how to structure your component's code.
+
+Simple, presentational components should go in the `src/components/UIKit`
+directory. More complex components should be in their own directory within
+`src/components`. Each component must be exported from `src/index.js`
```

Comment:
and have a test case in `exports.test.js` to verify that it is exported and usable externally.

Reply:
This is mentioned in the Tests subsection. I'll add bit more description there about the export test.